### DISCUSSION TITLE
External trigger (automation) -> app

### DISCUSSION THREAD

**Discussion by bjornclauw** [Budibase/budibase#9881] [Ideas]
Hi,

I know we have the ability to trigger an automation via a webhook or a row update, which then runs an asynchronous task on the server. (which is already pretty powerful) 
I was wondering if there are any plans (or if it is possible today) to enable the automation to directly interact with the app?

for example:
- trigger a datasource refresh
- change an appState variable (to update an icon color, to change a header value, ...)
- parse a received email and transfer the values to app
- parse data from a CRON automation (for example REST datasource) and send it to an appState variable/datasource (without refreshing)
- ...

**Comment by melohagan** [comment]
Hey @bjornclauw 

This is a great idea. We had discussed this internally not so long ago, although I wasn't able to dig out the discussion. 

I've raised a feature request from your suggestion! https://github.com/Budibase/budibase/issues/9890
